packages/better-auth/src/plugins/device-authorization/index.ts

```diff
@@ -0,0 +1,855 @@
+	 * Function to generate a device code.
+	 */
+	generateDeviceCode?: () => Promise<string> | string;
+	/**
+	 * Function to generate a user code.
+	 */
+	generateUserCode?: () => Promise<string> | string;
+	/**
+	 * Whether to format user codes with hyphens for readability.
+	 * @default true
+	 */
+	formatUserCode?: boolean;
+	/**
+	 * Enable rate limiting for token polling.
+	 * @default true
+	 */
+	enableRateLimiting?: boolean;
+}
+
+const deviceCodeSchema: AuthPluginSchema = {
+	deviceCode: {
+		fields: {
+			deviceCode: {
+				type: "string",
+				required: true,
+			},
+			userCode: {
+				type: "string",
+				required: true,
+			},
+			userId: {
+				type: "string",
+				required: false,
+			},
+			expiresAt: {
+				type: "date",
+				required: true,
+			},
+			status: {
+				type: "string",
+				required: true,
+				defaultValue: "pending",
+			},
+			lastPolledAt: {
+				type: "date",
+				required: false,
+			},
+			pollingInterval: {
+				type: "number",
+				required: false,
+			},
+			clientId: {
+				type: "string",
+				required: false,
+			},
+			scope: {
+				type: "string",
+				required: false,
+			},
+		} satisfies Record<string, FieldAttribute>,
+	},
+};
+
+export { deviceAuthorizationClient } from "./client";
+
+const DEVICE_AUTHORIZATION_ERROR_CODES = {
+	INVALID_DEVICE_CODE: "Invalid device code",
+	EXPIRED_DEVICE_CODE: "Device code has expired",
+	EXPIRED_USER_CODE: "User code has expired",
+	AUTHORIZATION_PENDING: "Authorization pending",
+	ACCESS_DENIED: "Access denied",
+	INVALID_USER_CODE: "Invalid user code",
+	DEVICE_CODE_ALREADY_PROCESSED: "Device code already processed",
+	POLLING_TOO_FREQUENTLY: "Polling too frequently",
+	USER_NOT_FOUND: "User not found",
+	FAILED_TO_CREATE_SESSION: "Failed to create session",
+	INVALID_DEVICE_CODE_STATUS: "Invalid device code status",
+	AUTHENTICATION_REQUIRED: "Authentication required",
+} as const;
+
+export const deviceAuthorization = (
+	options: DeviceAuthorizationOptions = {},
+) => {
+	const opts = {
+		expiresIn: 1800,
+		interval: 5,
+		deviceCodeLength: 40,
+		userCodeLength: 8,
+		userCodeCharset: "ABCDEFGHJKLMNPQRSTUVWXYZ23456789", // Excluding similar looking characters
+		formatUserCode: true,
+		enableRateLimiting: true,
+		...options,
+	};
+
+	const generateDeviceCode = async () => {
+		if (opts.generateDeviceCode) {
+			return opts.generateDeviceCode();
+		}
+		return generateRandomString(opts.deviceCodeLength, "a-z", "A-Z", "0-9");
+	};
+
+	const generateUserCode = async () => {
+		if (opts.generateUserCode) {
+			return opts.generateUserCode();
+		}
+		const chars = opts.userCodeCharset;
+		let code = "";
+		for (let i = 0; i < opts.userCodeLength; i++) {
+			code += chars[Math.floor(Math.random() * chars.length)];
+		}
+		return code;
+	};
+
+	const formatUserCode = (code: string) => {
+		// Format with hyphen for readability if enabled (e.g., "WDJB-MJHT")
+		if (opts.formatUserCode && code.length === 8) {
+			return `${code.slice(0, 4)}-${code.slice(4)}`;
```

Comment:
This is not good, need to remove `-` from the code side. It should be user-land defined behavior